### DISCUSSION TITLE
How do I use redirect in composables?

### DISCUSSION THREAD

**Discussion by kkfive** [nuxt/nuxt#21255] [Questions]
code url
https://stackblitz.com/edit/nuxt-starter-owpidf?file=composables%2FcustomFetch.ts%3AL22,pages%2Fuser%2Findex.vue

When the interface gives me 401 error, I want to redirect to the login page

please help me 

🙏🙏🙏🙏🙏🙏

**Comment by AasmundN** [comment]
I am also wondering about this.

**Comment by zubozrout** [comment]
Could adding in:
```
experimental: {
    asyncContext: true
}
```
under `defineNuxtConfig({ ... })` to `nuxt.config.ts` help here?

I was getting some weird not really helpful error without that:
```
Error: [nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at `https://nuxt.com/docs/guide/concepts/auto-imports#vue-and-nuxt-composables`.
```

... but adding in this config things started working for me fortunately.

**Comment by AasmundN** [comment]
Something like this worked for me.
```ts
await nuxtApp.runWithContext(()=> 
    navigateTo(...)
  )
)
```
